Flow.Launcher/ViewModel/MainViewModel.cs

```diff
@@ -1331,36 +1371,56 @@ private Query ConstructQuery(string queryText, IEnumerable<CustomShortcutModel>
                 queryBuilder.Replace('@' + shortcut.Key, shortcut.Expand());
             }
 
-            string customExpanded = queryBuilder.ToString();
+            // Applying builtin shortcuts
+            BuildQuery(builtInShortcuts, queryBuilder, queryBuilderTmp);
+
+            return QueryBuilder.Build(queryBuilder.ToString().Trim(), PluginManager.NonGlobalPlugins);
+        }
+
+        private void BuildQuery(IEnumerable<BaseBuiltinShortcutModel> builtInShortcuts,
+            StringBuilder queryBuilder, StringBuilder queryBuilderTmp)
+        {
+            var customExpanded = queryBuilder.ToString();
+
+            var queryChanged = false;
 
-            Application.Current.Dispatcher.Invoke(() =>
+            foreach (var shortcut in builtInShortcuts)
             {
-                foreach (var shortcut in builtInShortcuts)
+                try
                 {
-                    try
+                    if (customExpanded.Contains(shortcut.Key))
                     {
-                        if (customExpanded.Contains(shortcut.Key))
+                        string expansion;
+                        if (shortcut is BuiltinShortcutModel syncShortcut)
                         {
-                            var expansion = shortcut.Expand();
-                            queryBuilder.Replace(shortcut.Key, expansion);
-                            queryBuilderTmp.Replace(shortcut.Key, expansion);
+                            expansion = syncShortcut.Expand();
                         }
-                    }
-                    catch (Exception e)
-                    {
-                        App.API.LogException(ClassName,
-                            $"Error when expanding shortcut {shortcut.Key}",
-                            e);
+                        else if (shortcut is AsyncBuiltinShortcutModel asyncShortcut)
+                        {
+                            expansion = App.JTF.Run(() => asyncShortcut.ExpandAsync());
```

Comment:
let make it real async (or else it does not really makes sense to create an async shortcut)